src/main/java/com/google/devtools/build/lib/rules/python/PyCommon.java

```diff
@@ -158,16 +165,39 @@ public void addCommonTransitiveInfoProviders(RuleConfiguredTargetBuilder builder
    *     addSkylarkTransitiveInfo(PYTHON_SKYLARK_PROVIDER_NAME, createSourceProvider(...))
    */
   public static SkylarkClassObject createSourceProvider(
-      NestedSet<Artifact> transitivePythonSources, boolean isUsingSharedLibrary) {
+      NestedSet<Artifact> transitivePythonSources,
+      boolean isUsingSharedLibrary,
+      NestedSet<PathFragment> imports) {
     return NativeClassObjectConstructor.STRUCT.create(
         ImmutableMap.<String, Object>of(
             TRANSITIVE_PYTHON_SRCS,
             SkylarkNestedSet.of(Artifact.class, transitivePythonSources),
             IS_USING_SHARED_LIBRARY,
-            isUsingSharedLibrary),
+            isUsingSharedLibrary,
+            IMPORTS,
+            SkylarkNestedSet.of(String.class,
+                                makeImportStrings(imports ))
+            ),
         "No such attribute '%s'");
   }
 
+  /**
+   * Converts {@link NestedSet} of {@link PathFragment} to
+   * {@link NestedSet} of {@link String}
+   * @param imports - {@link NestedSet} of {@link PathFragment}
+   * @return - {@link NestedSet} of {@link String}
+   */
+  public static NestedSet<String> makeImportStrings(
+      NestedSet<PathFragment> imports) {
+    final NestedSetBuilder<String> builder = NestedSetBuilder.compileOrder();
+
+    for (PathFragment pathFragment : imports) {
+      builder.add(pathFragment.toString());
```

Comment:
Sorry for late reply. We do not plan to expose `PathFragment`s to Skylark. Can you do with `NestedSet<String>` instead?